controller: dcs-task: rename single-scan data member

adam1985d opened the pull request:
added a comment explaining the need for single-scan data members and the
need to differentiate between single-scan-requests and continuous-scans.

changed `m_is_single_scan` to `m_single_scan_request`.
changed `m_is_single_scan_pending` to `m_single_scan_request_pending`.

Signed-off-by: Adam Dov <adamx.dov@intel.com>

tomereli reviewed (commented):
Its good to have the comment but I don't get why you change the name from `m_single_scan` to `m_single_scan_request` - if it is true, doesn't it mean that we are currently running a single scan? If so, adding `_request` suffix is confusing.

vitalybu commented on controller/src/beerocks/master/tasks/dynamic_channel_selection_task.h:
```diff
@@ -92,8 +92,13 @@ class dynamic_channel_selection_task : public task, public std::enable_shared_fr
 
-    bool m_is_single_scan_pending = false;
-    bool m_is_single_scan         = false;
+    // The single scan request scan-parameters and scan-results are stored separately than the
```
```suggestion
    // The single scan request scan-parameters and scan-results are stored separately from the
```

vitalybu replied:
Are there really cases where both scans are triggered simultaneously?
Why don't we handle it by creating two instances of the DCS task?

adam1985d replied:
fixed the comment.

an instance is per interface and not per run
the task is created as part of agent-join and "lives" until agent gets disconnected

vitalybu reviewed (approved):
Have some logic related questions, but the renaming looks fine to me, so approving.

adam1985d commented:
> See, here I think it will be more clear if we name it as follows:
> 
> ```
> if (m_single_scan_request_pending) {
>   m_single_scan = true;
>   m_single_scan_request_pending = false;
> }
> ```

@kantera800 I agree with @tomereli on this one - what do you think?

kantera800 commented:
> > See, here I think it will be more clear if we name it as follows:
> > ```
> > if (m_single_scan_request_pending) {
> >   m_single_scan = true;
> >   m_single_scan_request_pending = false;
> > }
> > ```
> 
> @kantera800 I agree with @tomereli on this one - what do you think?

yes its better

adam1985d committed:
controller: dcs-task: rename single-scan data member

added a comment explaining the need for single-scan data members and the
need to differentiate between single-scan-requests and continuous-scans.

changed m_is_single_scan to m_single_scan_request.
changed m_is_single_scan_pending to m_single_scan_request_pending.

Signed-off-by: Adam Dov <adamx.dov@intel.com>
